.github/workflows/functional-bats.yml

```
@@ -0,0 +1,98 @@
+name: Functional Tests
+on:
+  workflow_call:
+    inputs:
+      tests-path:
+        required: false
+        type: string
+        default: 'tests/functional/testsuites/'
+      php-extensions:
+        required: false
+        type: string
+        default: 'autoload_psr-pprkut/autoload-psr@0.2.0'
+      php-ini-settings:
+        required: false
+        type: string
+        default: ''
+    secrets:
+      SSH_PRIVATE_KEY_FOR_CLONING_NONPUBLIC_BACKEND_GIT_REPOSITORIES:
+        required: true
+
+jobs:
+  functional-tests:
+    runs-on: ubuntu-latest
+    name: Install Bats and bats libs
+    permissions:
+      contents: read
+      issues: read
+      checks: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      
+      - name: Cache decomposer dependencies
+        id: cache-decomposer
+        uses: actions/cache@v4
+        with:
+          path: | 
+            ${{ github.workspace }}/vendor
+            ${{ github.workspace }}/decomposer.autoload.inc.php
+          key: ${{ runner.os }}-${{ hashFiles('decomposer.json') }}
+          restore-keys: 
+            ${{ runner.os }}-
+
+      - name: Decomposer checkout
+        uses: actions/checkout@v4
+        with:
+            repository: 'move-backend/decomposer'
+
+            # Relative path under $GITHUB_WORKSPACE to place the repository
+            path: 'decomposer'
+
+      - if: ${{ steps.cache-decomposer.outputs.cache-hit != 'true' }}
+        name: Prepare decomposer target directory
+        run: mkdir -p "$GITHUB_WORKSPACE/vendor"
+  
+      - name: Configure ssh keys
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: |
+            ${{ secrets.SSH_PRIVATE_KEY_FOR_CLONING_NONPUBLIC_BACKEND_GIT_REPOSITORIES }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.1
+          extensions: ${{ inputs.php-extensions }}
+          ini-values: include_path='.:/usr/share/php:$GITHUB_WORKSPACE/vendor', ${{ inputs.php-ini-settings }}
+
+      - name: Install dependencies
+        run: DECOMPOSER_TARGET_DIR="$GITHUB_WORKSPACE/vendor" decomposer/bin/decomposer install
+
+      - name: Setup Bats and bats libs
+        id: setup-bats
+        uses: bats-core/bats-action@3.0.0
+
+      - name: Run setup
+        run: ./support/setup.sh
+
+      - name: Set config overrides
+        run: |
+          cp -a config/conf.application.local.inc.php.sample config/conf.application.local.inc.php
+          sed -i "s|{{ environment }}|ci|" config/conf.application.local.inc.php
+
+      - name: Functional tests
+        shell: bash
+        env:
+          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
+          TERM: xterm
+        run: bats --recursive ${{ inputs.tests-path }} --print-output-on-failure --filter-tags '!mariadb:true' --report-formatter junit --output $PWD
```

This is more secure.
```suggestion
          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
          TERM: xterm
          TEST_PATH: ${{ inputs.tests-path }}
        run: bats --recursive "$TEST_PATH" --print-output-on-failure --filter-tags '!mariadb:true' --report-formatter junit --output $PWD
```